java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/verifier/VerifyBuiltArtifactsCommand.java

```diff
@@ -157,34 +231,43 @@ private void initMaven(Path globalSettingsFile, Path settingsFile) throws IOExce
             Log.debugf("Mirror for %s: %s", centralRepository.getId(), mirror);
             remoteRepositories = List.of(mirror);
         } else {
-            Log.debugf("Using repository URL %s", repositoryUrl);
-            remoteRepositories = List.of(new Builder("internal", "default", repositoryUrl).build());
+            Log.debugf("Using repository URL %s", options.mavenOptions.repositoryUrl);
+            remoteRepositories = List.of(new Builder("internal", "default", options.mavenOptions.repositoryUrl).build());
         }
     }
 
-    private boolean handleJar(Path file, String coords) {
+    private int handleJar(Path remoteFile, Path file, List<String> excludes) {
+        var left = new JarInfo(remoteFile);
+        var right = new JarInfo(file);
+        return left.diffJar(right, excludes);
+    }
+
+    private int handleJar(Path file, String coords, List<String> excludes) {
         try {
             var optionalRemoteFile = resolveArtifact(coords, remoteRepositories, session, system);
 
             if (optionalRemoteFile.isEmpty()) {
                 Log.warnf("Ignoring missing artifact %s", coords);
-                return false;
+                return 1;
             }
 
             var remoteFile = optionalRemoteFile.get();
-            var left = new JarInfo(remoteFile);
-            var right = new JarInfo(file);
             Log.infof("Verifying %s (%s, %s)", coords, remoteFile.toAbsolutePath(), file.toAbsolutePath());
-            var failed = left.diffJar(right);
-            Log.debugf("Verification of %s %s", coords, failed ? "failed" : "passed");
-            return failed;
+            var numFailures = handleJar(remoteFile, file, excludes);
+            Log.debugf("Verification of %s %s", coords, numFailures > 0 ? "failed" : "passed");
+
+            if (numFailures > 0) {
+                numFailures++;
+            }
+
+            return numFailures;
         } catch (OutOfMemoryError e) {
             //HUGE hack, but some things are just too large to diff in memory
             //but we would need a complete re-rewrite to handle this
             //these are usually tools that have heaps of classes shaded in
             //we just ignore this case for now
             Log.errorf(e, "Failed to analyse %s as it is too big", file);
-            return false;
```

Comment:
This should return 0. If we run out of memory we still need to be able to deploy it.

Reply:
Obviously, `false` should have been `0` and not `1`. I will fix it. The general idea was to return `1` for generic errors, and `1 + count` for an error list of size `count`.